File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    var answer = 0;
    const addTarget = (tmp, sum, res) => {
        if(numbers.length === tmp) {
            if(sum === target) {
                res += 1;
            }
            answer += res;
            return;
        }
        addTarget(tmp + 1, sum + numbers[tmp], res);
        addTarget(tmp + 1, sum - numbers[tmp], res);
    }
    
    addTarget(0, 0, answer);
    return answer;
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
from collections import deque

def solution(numbers, target):
    answer = 0
    q = deque()
    q.append([numbers[0], 0])
    q.append([(-1) * numbers[0], 0])
    while q:
        c_sum, c_idx = q.popleft()
        if c_idx == len(numbers) - 1:
            if c_sum == target:
                answer += 1
        else:
            if c_idx < len(numbers) - 1:
                q.append([c_sum + numbers[c_idx + 1], c_idx + 1])
                q.append([c_sum - numbers[c_idx + 1], c_idx + 1])
    return answer